pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1789,48 +1789,29 @@ func isDeviceOpened(deviceToDetach AttachedVolume, hostUtil mount.HostUtils) (bo
 	return deviceOpened, nil
 }
 
-func useCSIPlugin(vpm *volume.VolumePluginMgr, spec *volume.Spec) bool {
-	// TODO(#75146) Check whether the driver is installed as well so that
-	// we can throw a better error when the driver is not installed.
-	// The error should be of the approximate form:
-	// fmt.Errorf("in-tree plugin %s is migrated on node %s but driver %s is not installed", pluginName, string(nodeName), driverName)
+func useCSIPlugin(vpm *volume.VolumePluginMgr, spec *volume.Spec, nodeName *types.NodeName) bool {
 	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIMigration) {
 		return false
 	}
 	if csilib.IsPVMigratable(spec.PersistentVolume) || csilib.IsInlineMigratable(spec.Volume) {
 		migratable, err := vpm.IsPluginMigratableBySpec(spec)
 		if err == nil && migratable {
-			return true
+			// nodeName would be nil for mount operations where check for driver is not applicable
+			if nodeName == nil {
+				return true
+			}
+			// we are only interested in whether the driver is installed. If the
+			// driver is not installed, err would not be nil
+			if _, err = checkDriverInstalled(vpm, spec, *nodeName); err == nil {
```

Comment:
this changes behavior significantly in that now if driver is not installed we do not do migration. In fact, when Migration flags are on we **always** want to do migration whether the driver is installed or not.